cpp/src/arrow/compute/kernels/codegen_internal_test.cc

```diff
@@ -159,6 +159,12 @@ TEST(TestDispatchBest, CommonTemporal) {
   args = {timestamp(TimeUnit::SECOND, "America/Phoenix"),
           timestamp(TimeUnit::SECOND, "UTC")};
   ASSERT_EQ(CommonTemporal(args.data(), args.size()), nullptr);
+
```

Comment:
Please add some test cases to check (for example) that there is no common type between time and date

Reply:
```c++
  args = {date32(), time32(TimeUnit::SECOND)};
  ASSERT_EQ(CommonTemporal(args.data(), args.size()), nullptr);

  args = {timestamp(TimeUnit::SECOND), time32(TimeUnit::SECOND)};
  ASSERT_EQ(CommonTemporal(args.data(), args.size()), nullptr);
```

Have added